src/etc/inc/services.inc

```diff
@@ -789,6 +789,9 @@ EOPP;
 			}
 		}
 
+		// instantiate class before pool definition
+		$dhcpdconf .= "class \"s_{$dhcpif}\" {\n	match hardware;\n}\n";
```

Comment:
What if the MAC is empty? Entries can be specified based on MAC or Client ID. Either one could be empty.

The `dhcpd.conf` man page has an example that might work:

    match pick-first-value	(option	dhcp-client-identifier,	hardware);

Reply:
done